modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsImpl.java

```diff
@@ -188,6 +224,98 @@ public void decrementIndexBuildCountPartitionsLeft() {
         idxBuildCntPartitionsLeft.decrement();
     }
 
+    /**
+     * Number of rebalanced partitions left.
+     */
+    public long getRebalancingPartitionsLeft() {
+        return rebalancingPartitionsLeft.value();
+    }
+
+    /**
+     * Add rebalancing partitions.
+     *
+     * @param numPartitions Number of partitions to rebalance.
+     */
+    public void addRebalancingPartitions(long numPartitions) {
+        rebalancingPartitionsLeft.add(numPartitions);
+    }
+
+    /**
+     * Сancel rebalancing partitions.
+     *
+     * @param numPartitions Number of cancelled partitions.
+     */
+    public void cancelRebalancingPartitions(long numPartitions) {
+        rebalancingPartitionsLeft.add(-numPartitions);
+    }
+
+    /**
+     * Rebalance partition callback.
+     */
+    public void onPartitionRebalanced() {
+        rebalancingPartitionsLeft.decrement();
+    }
+
+    /**
+     * Rebalance entry store callback.
+     */
+    public void onRebalanceKeyReceived() {
+        rebalancingReceivedKeys.increment();
+    }
+
+    /**
+     * Rebalanced keys count.
+     */
+    public long getRebalancingReceivedKeys() {
+        return rebalancingReceivedKeys.value();
+    }
+
+    /**
+     * Rebalanced bytes count.
+     */
+    public long getRebalancingReceivedBytes() {
```

Comment:
Do we need a `getters` at all? It seems all the metrics can be obtained from the metric registry.